test/corpus/aliases.txt

```diff
@@ -0,0 +1,103 @@
+================================================================================
+Type alias
+================================================================================
+
+alias Foo = HelloWorld
+
+--------------------------------------------------------------------------------
+
+(source_file
+  (alias (type_definition (type_identifier))
+   (type_definition (type_identifier))))
+
+================================================================================
+Type alias with generics
+================================================================================
+
+alias Foo<a> = HelloWorld<a>
+
+--------------------------------------------------------------------------------
+
+(source_file
+  (alias
+    (type_definition
+      (type_identifier)
+      (type_argument
+        (identifier)))
+    (type_definition
+      (type_identifier)
+      (type_argument
+        (identifier)))))
+
+================================================================================
+Type alias with type dependencies
+================================================================================
+
+alias Foo<Int> = HelloWorld<Int>
+
+--------------------------------------------------------------------------------
+
+(source_file
+  (alias
+    (type_definition
+      (type_identifier)
+      (type_argument
+        (type_definition
+          (type_identifier))))
+    (type_definition
+      (type_identifier)
+      (type_argument
+        (type_definition
+          (type_identifier))))))
+
+================================================================================
+Type alias with multiple type dependencies
+================================================================================
+
+alias Foo<Int, String> = HelloWorld<Int, String>
```

Comment:
This is wrong, we shouldn't expect upper case names on the left

Reply:
For reference

![image](https://github.com/aiken-lang/tree-sitter-aiken/assets/12070598/66fd816a-679d-47b6-a7eb-c04d7b2fb6e5)

Reply:
Nvm we decided this is better for syntax highlighting